skorch/dataset.py

```diff
@@ -75,6 +89,41 @@ def uses_placeholder_y(ds):
     return isinstance(ds, Dataset) and hasattr(ds, "y") and ds.y is None
 
 
+ERROR_MSG_1_ITEM = (
+    "You are using a non-skorch dataset that returns 1 value. "
+    "Remember that for skorch, Dataset.__getitem__ must return exactly "
+    "2 values, X and y (more info: "
+    "https://skorch.readthedocs.io/en/stable/user/dataset.html).")
+
+
+ERROR_MSG_MORE_THAN_2_ITEMS = (
+    "You are using a non-skorch dataset that returns {} values. "
+    "Remember that for skorch, Dataset.__getitem__ must return exactly "
+    "2 values, X and y (more info: "
+    "https://skorch.readthedocs.io/en/stable/user/dataset.html).")
+
+
+def unpack_data(data):
```

Comment:
Exposing this as an argument in `NeuralNet` may give the flexibility needed to control how data is "unpacked".

Reply:
I was wondering about that too. Is there a case where you would want to change this specific behavior? My impression is that the assumption is so fundamental that you would never change it anyway, but perhaps I'm missing a case.

Reply:
The use case would be if one uses a pure PyTorch Dataset where its `__getitem__` function returns more than 2 values. A custom `unpack_data` would transform the data to a 2-tuple. 

I think this is really edge case. Advising a user to adjust the `__getitem__` to return a 2-tuple as  in https://github.com/dnouri/skorch/issues/422 would be better.

It is probably better not to expose `unpack_data`.

Reply:
It is also pretty easy to define your own data iterator (e.g. a generator) which could do the altered unpacking. I'm against exposing `unpack_data` as well.

Reply:
Okay, then we leave it as is.